### ISSUE TITLE
Will we need a QUIC adaptation layer for RTP/RTCP, similar to the one defined for HTTP/3?

### ISSUE THREAD

**Issue by SpencerDawkins** [SpencerDawkins/sdp-rtp-quic#2]
* When QUIC establishes connections, it uses IP addresses but then expects applications to use connection IDs to refer to connections, even if the underlying IP addresses change because of NAT binding, and even if the QUIC implementation performs QUIC connection migration itself, so the underlying IP addresses change. RTP applications expect to use IP addresses, not QUIC connection IDs. Must we specify an RTP/RTCP adaptation layer, similar to {{I-D.ietf-quic-http}} for HTTP/3?

**Comment by juberti** [comment]
This is pretty interesting, because I can imagine a lot of applications not wanting to use IP addresses if QUIC is going to take care of all the details. I'll have to think about what this might mean in practice.

**Comment by SpencerDawkins** [comment]
@juberti - a lot of us need to think about this! 

I'm not the wizard of QUIC, but offhand, I can think of both connection migration and multipath that would expect an application to understand QUIC connection IDs, and wouldn't work if the application thinks it's only talking to one source-destination IP address pair, and at that point, I stopped trying to think of more things that wouldn't work. :fearful:

**Comment by juberti** [comment]
why does the application need to care though? It can just pretend it's talking to a single IP:port over a virtual connection (similar to how ICE is handled) and let the quic implementation handle all the details.

**Comment by SpencerDawkins** [comment]
@juberti, I agree with your statement, and ISTM that we need to talk about this when we're talking about the functionality of the "RTP over QUIC" encapsulation layer. 

I was using the term **"QUIC adaptation layer for RTP/RTCP"** to describe something on top of the QUIC implementation (if the "QUIC implementation" is roughly doing [RFC 9000](https://www.rfc-editor.org/rfc/rfc9000.html) plus extensions), similar to [HTTP/3](https://datatracker.ietf.org/doc/draft-ietf-quic-http/). Does that make sense?

**Comment by juberti** [comment]
Sure, we definitely need an adaptation layer. I'm not sure if that's distinct from rtp-over-quic, but yes, somewhere we need to explain all the nuances of how sending over QUIC is different than sending over straight UDP. I think most of the actual details related to IP addresses will be pretty straightforward though.
